projects/public-search/src/app/document-detail/request/request.component.ts

```diff
@@ -50,25 +57,35 @@ export class RequestComponent implements OnInit {
   /**
    * Constructor
    * @param _itemApiService - ItemApiService
+   * @param _holdingsApiService: HoldingsApiService
    * @param _userService - UserService
    */
   constructor(
     private _itemApiService: ItemApiService,
-    private _userService: UserService
+    private _holdingsApiService: HoldingsApiService,
+    private _userService: UserService,
   ) { }
 
   /** OnInit hook */
   ngOnInit(): void {
+    if (this.recordComponent === 'holding') {
+      this.apiRequest = this._holdingsApiService;
+    }else if (this.recordComponent === 'item') {
+      this.apiRequest = this._itemApiService;
+    }
+
```

Comment:
* `apiRequest` doesn't need to be component attribute, you could use simple function variable because it's only used into this function.
* a switch statement seems more elegant in this case 
```javascript
let apiRequest = null;
switch(this.recordComponent) {
    case 'holding': {apiRequest = this._holdingsApiService; break; }
    case 'item': { apiRequest = this._holdingsApiService; break; }
    default: throw new TypeError(this.recordComponent + 'isn\'t supported');
}
```
or a ternary instruction (seems suffisent at this time)
```javascript
const apiRequest = (this.recordComponent === 'holding') ? this._holdingsApiService : this._itemApiService;
```